From 8d56bb9661eb725983413dc4cfae272e36e6da9f Mon Sep 17 00:00:00 2001
From: donghuawei <donghuawei@eswincomputing.com>
Date: Wed, 16 Apr 2025 19:48:47 +0800
Subject: [PATCH 491/652] feat: vpu7702 smmu cfg modify

Changelogs:
	smmu cfg dts, dma-ranges modify iova size for support large
	iova

Signed-off-by: donghuawei <donghuawei@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7700-vpu7702.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-vpu7702.dts b/arch/riscv/boot/dts/eswin/eic7700-vpu7702.dts
index 4134802ddc5e..0a54b5c27bb6 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-vpu7702.dts
+++ b/arch/riscv/boot/dts/eswin/eic7700-vpu7702.dts
@@ -184,12 +184,13 @@ d0_ipcm:ipcm@0 {
 			memory-region = <&ipcm_vdev0vring0>,<&ipcm_vdev0vring1>,<&ipcm_vdev0buffer>;
 			interrupt-parent = <&plic0>;
 			interrupts = <119>;
+		  	smmu_dev = <&smmu_cfg_0>;
 		};
 
 		dev_smmu_cfg: smmu_cfg@0 {
 			compatible = "eswin,vpu7702_dev_smmu_cfg";
 			#size-cells = <2>;
-			dma-ranges = <0x0 0x20000000 0x0 0xc0000000 0x0 0x40000000>;
+			dma-ranges = <0x0 0x20000000 0x0 0xc0000000 0x0 0xc0000000>;
 			iommus = <&smmu0 WIN2030_SID_DEV_FOO_B>;
 			iommu-map = <0x000000 &smmu0 0xff0000 0xffffff>;
 			tbus = <WIN2030_TBUID_PCIE>;
-- 
2.54.0

